docs/design.md

```diff
@@ -0,0 +1,216 @@
+- Local PV Provisioner is used to create Local PVs representing the disks. The disks that have be used by the Local PV Provisioner have to be mounted into a directory (like /mnt/localdisks) and this location should be passed using a Config Map to the Local PV Provisioner. 
+
+- Modify/extend the provisioner to add the following additional information:
+  * PVs will be associated with node,rack information that will be obtained by the labels in the nodes. (TODO Kiran - Find the information on how nodes are associated with rack/zone information)
+  * PVs will get additional information regarding the type of the disk, iops that is capable of etc., 
+
+- Storage Solution Operators (say OpenEBS cStor Pool Provisioner or GlusterFS Provisioner) could then:
+  * query for the PVs using `kubectl get pv` (filtered with local-provisioner type) to get information on all the disks in the Cluster.
+  * create the storage pods by passing/binding them with the Local PVs. 
+
+- Storage Solution Operators (say OpenEBS Operator or GlusterFS Operator) will have to monitor for faults on the PVs and send notification to the respective storage pods to take corrective actions. The monitoring includes information like:
+  * gathering smart statistics of the disk
+  * probing for mode-sense data on disks or listening for events generated by the underlying disk
+
+However with this Local PV approach, any of the following operations will require a storage pod restart (Issue https://github.com/kubernetes/kubernetes/issues/58569 ):
+  * new PVs have to attached to expand the storage available
+  * Remove or replace a failed PV
+
+A storage pod restarts can be expensive in terms of requiring data rebuilding and also introducing additional scenarios for failure combinations that could sometimes result in applications seeing high latencies or sometimes causing the volumes to get into read-only. 
+
+
+### Using node-disk-manager
+
+The workflow with _*node-disk-manager*_ would be to integrate into current approach of container attached storage solutions where - disks are accessed by mounting a single volume "/dev", while the _*node-disk-manager*_ will provide a generic way to:
+* manage (discover, provision and monitor) disks
+* manage storage pools (aggregate of disks)
+
+- node-disk-manager can be configured to run on all node or select nodes (designated as Storage Nodes) 
+
+- The operator that installs the node-disk-manager will also insert/create CRDs for Disk and DiskClaim CRDs
+
+- On start, the node-disk-manager will discover the disks attached to the node where it is running and will create Disk CRs if they were not already present. 
+
+- A Disk CR will contain information like:
+  * type
+  * resource identifier
+  * physical location - node, rack, zone
+  ```
+  apiVersion: openebs.io/v1
```

Comment:
probably we should update this with the latest CR we have currently.

Reply:
good point!